### PR TITLE
Added LSP tests for JSON encoder and decoder

### PR DESCRIPTION
**Pull request by kpadmasola**

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -3,12 +3,12 @@ packages: .
 source-repository-package
     type: git
     location: https://github.com/luc-tielen/llvm-codegen.git
-    tag: 42043c279797a4f57e254fcf5e60f3df56f9bf65
+    tag: c97de620ed147388cb6213864d33c63d199561e1
```
Comment (luc-tielen) [inline cabal.project:6]: Why are these pinned commits updated?

File: tests/eclair/Test/Eclair/LSP/JSONSpec.hs [hunk 2]
```diff
@@ -46,7 +111,60 @@ spec = describe "LSP JSON processing" $ parallel $ do
...
+            "type": "hover",
+            "command": {
+              "position": {"line": 100, "column": 22},
+              "file": "/tmp/file.eclair"
+            }
+          }
+        |]
+        (Hover "/tmp/file.eclair" (SourcePos 100 22))
+
+    it "can decode a command from JSON" $ do
```
Comment (luc-tielen) [inline tests/eclair/Test/Eclair/LSP/JSONSpec.hs:128]: Use unique names for the test names. Makes it easier to debug when 1 test fails.
For example: `can decode a document-highlight command from JSON` (also do this for other test names)

Comment (luc-tielen) [reply tests/eclair/Test/Eclair/LSP/JSONSpec.hs:128]: Alternatively you can just group them all together under 1 `it` in this case (should not change that much anyway..)

**Review by luc-tielen — CHANGES_REQUESTED** [review]

File: cabal.project (hunk 1 above)
Comment (kpadmasola) [reply cabal.project:6]: Build was failing for me on MacOS as it was looking for LLVM-17 which is not available on my machine. The updated commit of llvm-codegen supports fallback to lower versions.

Comment (luc-tielen) [reply cabal.project:6]: Right, I forgot to bump that.. thank you :smile:

File: tests/eclair/Test/Eclair/LSP/JSONSpec.hs (hunk 2 above)
Comment (kpadmasola) [reply tests/eclair/Test/Eclair/LSP/JSONSpec.hs]: I modified the test names as suggested, thanks.

**Comment by luc-tielen** [comment]
Thanks!

**Pull request merged**